bin/reth/src/node/mod.rs

```diff
@@ -456,17 +457,41 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
         let mut hooks = EngineHooks::new();
 
         let pruner_events = if let Some(prune_config) = prune_config {
-            info!(target: "reth::cli", ?prune_config, "Pruner initialized");
+            let mut segments = SegmentSet::new();
+
+            if let Some(mode) = prune_config.segments.receipts {
+                segments = segments.add_segment(reth_prune::segments::Receipts::new(mode));
+            }
+            if !prune_config.segments.receipts_log_filter.is_empty() {
+                segments = segments.add_segment(reth_prune::segments::ReceiptsByLogs::new(
```

Comment:
could we combine this with the `prune_config` call?
or at least move to a separate function?

Reply:
yep, moved to a separate `build_pruner` method akin to pipeline building